common/test/com/commercetools/sunrise/common/template/engine/handlebars/PlayJavaFormResolverTest.java

```diff
@@ -0,0 +1,52 @@
+package com.commercetools.sunrise.common.template.engine.handlebars;
+
+import com.commercetools.sunrise.common.forms.ErrorBean;
+import com.commercetools.sunrise.common.forms.ErrorsBean;
+import com.commercetools.sunrise.common.utils.ErrorFormatter;
+import com.google.common.collect.Maps;
+import org.junit.Test;
+import org.mockito.Mockito;
+import play.data.Form;
+import play.data.validation.ValidationError;
+
+import java.util.*;
+import java.util.function.Predicate;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PlayJavaFormResolverTest {
+
+    PlayJavaFormResolver formResolver = new PlayJavaFormResolver(singletonList(Locale.ENGLISH), new ErrorFormatter() {
+        @Override
+        public String format(List<Locale> locales, String message) {
+            return message;
+        }
+    });
+
+    @Test
+    public void extractErrors() throws Exception {
+        String errorField1 = "field1";
+        String errorField2 = "field2";
+        Form form = Mockito.mock(Form.class);
+        Map<String, List<ValidationError>> errorMap = new HashMap<String, List<ValidationError>>();
+        errorMap.put(errorField1, Arrays.asList(new ValidationError("errorkey1", "errorMessage1"),
+                new ValidationError("errorkey2", "errorMessage2")));
+        errorMap.put(errorField2, Collections.singletonList(new ValidationError("errorkey21", "errorMessage21")));
+
+        Mockito.when(form.errors()).thenReturn(errorMap);
```

Comment:
I would extract all this:
```java
Form form = Mockito.mock(Form.class);
Map<String, List<ValidationError>> errorMap = new HashMap<String, List<ValidationError>>();
errorMap.put(errorField1, Arrays.asList(new ValidationError("errorkey1", "errorMessage1"),
                new ValidationError("errorkey2", "errorMessage2")));
errorMap.put(errorField2, Collections.singletonList(new ValidationError("errorkey21", "errorMessage21")));
Mockito.when(form.errors()).thenReturn(errorMap);
```
into a method, called for example `formWithSomeErrorsForFields(field1, field2)`
This way the test is keeping only the important parts, with all boilerplate elsewhere.

Reply:
694feaf